### ISSUE TITLE
cryptopp.com website is down

### ISSUE THREAD

**Issue by branislavjenco** [weidai11/cryptopp#1022]
Hi, just thought I'd notify - it looks like cryptopp.com is down, maybe a DNS issue?

**Comment by noloader** [comment]
Thanks @branislavjenco.

Yeah, I can't get to it either. Let me ping Wei.

**Comment by jhihn** [comment]
Same here. USA/East coast

**Comment by noloader** [comment]
@branislavjenco, @jhihn,

I talked to Wei. He said I should find another hosting provider and restore from backup.

I'm **_guessing_** something went sideways with GoDaddy. GoDaddy has been the host for about 15 years.

Here is some of the back story, but I don't know if it has something to do with the current problem... In the past, GoDaddy sent us nasty emails saying we were using too many resources and threatening to cancel service. We tracked it down to spam bots hitting the wiki and causing cpu spikes. Each time the a spam bot tries to make a spam page, Mediawiki spins up the New Page editor. The New Page editor caused a spike in cpu usage. The pages were not saved because the spam bot did not have an account. At peak times we would get 3 to 5 spam hits per second.

**Comment by noloader** [comment]
Hi Everyone,

We are cutting over to a new VPS provider. The website and manual are restored (easy). Next is the wiki and its database (hard).

You can get the the site by IP at the moment: http://198.251.77.197/.

I asked Wei to cutover DNS to the new VM.

Jeff

**Comment by jhihn** [comment]
Nope, http://198.251.77.197/ is giving me a connection reset error

**Comment by jhihn** [comment]
No longer erroring.

**Comment by noloader** [comment]
@jhihn,

Last night around midnight the server lost networking. I'm not sure why:

![systemd-wtf](https://user-images.githubusercontent.com/3538226/113189430-3080e480-9229-11eb-8251-f1a062455db7.png)

I running through another restore right now. It should take about an hour.

**Comment by noloader** [comment]
If you are interested, here is the backup script we use for the VM. It is run by a Systemd service and triggered by a timer:

* https://github.com/weidai11/website/blob/master/systemd/bitvise-backup

And here is the restore process. Most of it could be automated, but we have only needed to do it 3 times in the last 20 years or so.

* https://github.com/weidai11/website/blob/master/restore/restore.txt

Finally, here is how we setup the VM. Apache and PHP are the ones that take a little extra time.

* https://github.com/weidai11/website/blob/master/apache-php/apache-php.txt

**Comment by sebres** [comment]
It happens again... got connection timeout over http, ping looks like this:
```
$ ping www.cryptopp.com
PING cryptopp.com (198.251.77.197) 56(84) bytes of data.
^C
--- cryptopp.com ping statistics ---
11 packets transmitted, 0 received, 100% packet loss, time 10241ms
```
